hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/systemcontracts/hts/cancelairdrops/TokenCancelAirdropTranslatorTest.java

```diff
@@ -0,0 +1,301 @@
+import static com.hedera.node.app.service.contract.impl.test.exec.systemcontracts.CallAttemptHelpers.prepareHtsAttemptWithSelectorForRedirectWithConfig;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.BDDMockito.given;
+
+import com.hedera.hapi.node.base.AccountID;
+import com.hedera.hapi.node.transaction.TransactionBody;
+import com.hedera.node.app.service.contract.impl.exec.gas.DispatchType;
+import com.hedera.node.app.service.contract.impl.exec.gas.SystemContractGasCalculator;
+import com.hedera.node.app.service.contract.impl.exec.scope.HederaNativeOperations;
+import com.hedera.node.app.service.contract.impl.exec.scope.VerificationStrategies;
+import com.hedera.node.app.service.contract.impl.exec.scope.VerificationStrategy;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.AddressIdConverter;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.DispatchForResponseCodeHtsCall;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.HtsCallAttempt;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.cancelairdrops.TokenCancelAirdropDecoder;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.cancelairdrops.TokenCancelAirdropTranslator;
+import com.hedera.node.app.service.contract.impl.hevm.HederaWorldUpdater.Enhancement;
+import com.hedera.node.config.data.ContractsConfig;
+import com.swirlds.config.api.Configuration;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class TokenCancelAirdropTranslatorTest {
+
+    @Mock
+    private TokenCancelAirdropDecoder decoder;
+
+    @Mock
+    private HtsCallAttempt attempt;
+
+    @Mock
+    private ContractsConfig contractsConfig;
+
+    @Mock
+    private Configuration configuration;
+
+    @Mock
+    private Enhancement enhancement;
+
+    @Mock
+    private AddressIdConverter addressIdConverter;
+
+    @Mock
+    private VerificationStrategies verificationStrategies;
+
+    @Mock
+    private VerificationStrategy verificationStrategy;
+
+    @Mock
+    private SystemContractGasCalculator gasCalculator;
+
+    @Mock
+    private HederaNativeOperations nativeOperations;
+
+    @Mock
+    private TransactionBody transactionBody;
+
+    @Mock
+    private AccountID payerId;
+
+    private TokenCancelAirdropTranslator subject;
+
+    @BeforeEach
+    void setUp() {
+        subject = new TokenCancelAirdropTranslator(decoder);
+    }
+
+    @Test
+    void matchesHTSCancelAirdropEnabled() {
```

Comment:
How about at least one test for an non-matching selector, when enabled?  (Obviously can't be comprehensive, just one mismatch will do.)

Reply:
Added